models/camelot/arbitrum/camelot_v3_arbitrum_trades.sql

```diff
@@ -0,0 +1,121 @@
+{{  config(tags=['dunesql'],
+    schema = 'camelot_v3_arbitrum',
+    alias = alias('trades'),
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index'])
+}}
+
+
+{% set project_start_date = '2022-06-14' %}
+{% set blockchain = 'arbitrum' %}
+
+WITH dexs AS (
+  SELECT
+    swaps.evt_block_time AS block_time,
+    swaps.recipient AS taker,
+    CAST(NULL as VARBINARY) AS maker,
+    CASE
+      WHEN swaps.amount0 < INT256 '0' THEN abs(swaps.amount0)
+      ELSE abs(swaps.amount1)
+    END AS token_bought_amount_raw,
+    CASE
+      WHEN swaps.amount0 < INT256 '0' THEN abs(swaps.amount1)
+      ELSE abs(swaps.amount0)
+    END AS token_sold_amount_raw,
+    NULL AS amount_usd,
+    CASE
+      WHEN amount0 < INT256 '0' THEN pools.token0
+      ELSE pools.token1
+    END AS token_bought_address,
+    CASE
+      WHEN amount0 < INT256 '0' THEN pools.token1
+      ELSE pools.token0
+    END AS token_sold_address,
+    swaps.contract_address AS project_contract_address,
+    swaps.evt_tx_hash AS tx_hash,
+    swaps.evt_index
+  FROM 
+    {{ source('camelot_v3_arbitrum', 'AlgebraPool_evt_Swap') }} AS swaps
+  INNER JOIN 
+    {{ source('camelot_v3_arbitrum', 'AlgebraFactory_evt_Pool') }} AS pools
+      ON swaps.contract_address = pools.pool
+  {% if is_incremental() %}
+  WHERE swaps.evt_block_time >= date_trunc('day', now() - interval '7' day)
```

Comment:
We can use `incremental_predicate` to rewrite these incremental filters:
```suggestion
  WHERE {{incremental_predicate('swaps.evt_block_time')}}
```